### PR TITLE
Adding a page to the Guides about PR Reviews

### PR DESCRIPTION
**Pull request by Dayvis7**
## Summary of changes

This first draft of the webpage includes a checklist of information that PR reviewers should keep in mind while reviewing a PR. Along with the checklist, there are links to other PR reviewing resources. This is the first draft and is meant to leave room for more templates and other information that becomes necessary.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Associated Issues and PRs

- Issue: #184 

## Associated Developers

- Dev: @katyhuff

### CODE REVIEW

File: manual/guides/pullrequests.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+---
+layout: manual
+title: Pull Requests
+subtitle: "What to check in a PR Review"
+permalink: /manual/guides/pullrequests
+---
+
+# Reviewing a Pull Request
+
+- [Recommendations from Cyclus PRs](https://fuelcycle.org/kernel/pr_review.html)
```
Comment (katyhuff) [inline manual/guides/pullrequests.md:10]:
```suggestion
- [Cyclus PR Review Recommendations](https://fuelcycle.org/kernel/pr_review.html)
```

File: manual/guides/pullrequests.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
```
Comment (katyhuff) [inline manual/guides/pullrequests.md:11]:
```suggestion
```

File: manual/guides/pullrequests.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
+---
+layout: manual
+title: Pull Requests
```
Comment (katyhuff) [inline manual/guides/pullrequests.md:3]: 1) I think this file should be called pull_requests.md
2) I would expect a page titled "Pull Requests" to include both how to make PRs and how to review them. Possibly, it may be also helpful to add a couple of sentences on how to respond to PR review comments (e.g. how to deal with the suggesting functionality by committing the changes). 

If you'd like Reviewing a PR to be it's own page, that would be fine. If you'd like it to retain this title "Pull Requests," then the Making a Pull Request section should include links to existing guidance on how to make a PR. Guidance for making PRs can be found in my book and on the github help pages. Specific to this group, I would like to add that a PR in the group should:

- have GitHub labels
- have a GitHub project
- have a description in GitHub
- List requested reviewers (not everyone, but not no one. One or two PR reviewers is ideal. More can be the right choice sometimes. )
- be linked to an issue.
- be small (atomic, do just one thing)

File: manual/guides/pullrequests.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+permalink: /manual/guides/pullrequests
+---
+
+# Reviewing a Pull Request
+
+- [Recommendations from Cyclus PRs](https://fuelcycle.org/kernel/pr_review.html)
+
+- [Common mistakes and errors](https://blog.scottnonnenberg.com/top-ten-pull-request-review-mistakes/)
+
+Below is a checklist that lists many aspects that reviews should
```
Comment (katyhuff) [inline manual/guides/pullrequests.md:14]:
```suggestion
Below is a checklist the PR reviewer should
```

File: manual/guides/pullrequests.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+consider before approving a pull request. As a reminder, the
+[ARFC Code of Conduct](http://arfc.github.io/manual/coc) still applies
+in pull request reviews. It's important to practice proper reviewing
```
Comment (katyhuff) [inline manual/guides/pullrequests.md:17]:
```suggestion
in pull request reviews. Reviews should be both thorough and respectful. 
Critical feedback is encouraged but should be delivered with kindness.
```

File: manual/guides/pullrequests.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+while maintaining respect.
```
Comment (katyhuff) [inline manual/guides/pullrequests.md:18]:
```suggestion
```

File: manual/guides/pullrequests.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+```markdown
+- [ ] Read the PR description
+- [ ] Read through all the changes, considering the following questions.
+  - [ ] Is there anything you can praise about this PR? Start with
+        praise.
+  - [ ] Are variable names brief but descriptive?
+  - [ ] Are new/changed functions no longer than a paragraph?
+  - [ ] Do all function parameters have default values where appropriate?
+  - [ ] Is the code clear and clean? (see Robert C. Martin's Clean Code)
```
Comment (katyhuff) [inline manual/guides/pullrequests.md:28]: add link to the Clean Code book.

File: manual/guides/pullrequests.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  - [ ] Is there enough documentation?
+  - [ ] Does the programming style meet the requirements of the
+        repository (PEP8 for python, google for C++, etc.)
```
Comment (katyhuff) [inline manual/guides/pullrequests.md:31]: link to PEP8, link to google C++ style guide.

File: manual/guides/pullrequests.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+        added to confirm good behavior?
+  - [ ] Does the test actually test the new/changed functionality?
+  - [ ] Does the test successfully test edge cases?
+  - [ ] Does the test successfully test corner cases?
+  - [ ] If the repository has continuous integration: does the PR pass
+        the tests?
+  - [ ] If there is no continuous integration, check out the branch
+        locally, build, and run the tests.
+  - [ ] Do the tests pass on your machine?
+  - [ ] Is the PR free of random cruft (built files, SWP files, etc.)?
```
Comment (katyhuff) [inline manual/guides/pullrequests.md:42]:
```suggestion
  - [ ] Is the PR free of random cruft (built files, `.swp` files, etc.)?
```

File: manual/guides/pullrequests.md [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+  - [ ] Do all files in the PR belong in the repository?
+  - [ ] If the PR deletes files, is this appropriate?
+  - [ ] If the PR adds files or data, are these new files compatible
+        with the repository license?
+  - [ ] Does this PR close an issue? If so, be sure to descriptively
+        close this issue when the PR is merged.
+  - [ ] Make a review, leaving kind comments and suggesting changes
+        where needed (to resolve the above).
```
Comment (katyhuff) [inline manual/guides/pullrequests.md:50]:
```suggestion
        where needed (to resolve the above).
  - [ ] Has the author resolved all of the comments and suggestions in your review?
```

File: manual/guides/pullrequests.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+- [Recommendations from Cyclus PRs](https://fuelcycle.org/kernel/pr_review.html)
+
+- [Common mistakes and errors](https://blog.scottnonnenberg.com/top-ten-pull-request-review-mistakes/)
+
+Below is a checklist that lists many aspects that reviews should
+consider before approving a pull request. As a reminder, the
+[ARFC Code of Conduct](http://arfc.github.io/manual/coc) still applies
+in pull request reviews. It's important to practice proper reviewing
+while maintaining respect.
+
```
Comment (katyhuff) [inline manual/guides/pullrequests.md:19]:
```suggestion

If you are about to review a PR, consider copying and pasting 
this checklist into your review comment and checking the boxes 
as you go along. 

```

File: manual/guides/pullrequests.md [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+        the tests?
+  - [ ] If there is no continuous integration, check out the branch
+        locally, build, and run the tests.
+  - [ ] Do the tests pass on your machine?
+  - [ ] Is the PR free of random cruft (built files, SWP files, etc.)?
+  - [ ] Do all files in the PR belong in the repository?
+  - [ ] If the PR deletes files, is this appropriate?
+  - [ ] If the PR adds files or data, are these new files compatible
+        with the repository license?
+  - [ ] Does this PR close an issue? If so, be sure to descriptively
```
Comment (katyhuff) [inline manual/guides/pullrequests.md:47]: Let's actually move this one all the way to the end of the list (after "When you approve... " and before "Thank the author...")

File: manual/guides/pullrequests.md (hunk 7 above)
Comment (Dayvis7) [reply manual/guides/pullrequests.md:31]: Should I use a link to a free PDF that I found or something like the Amazon page where you can buy the book?

Comment (katyhuff) [reply manual/guides/pullrequests.md:31]: The UIUC library or the o'reilly access linked there:

- https://i-share-uiu.primo.exlibrisgroup.com/discovery/fulldisplay?docid=alma99944155312205899&context=L&vid=01CARLI_UIU:CARLI_UIU&tab=LibraryCatalog&lang=en
- https://learning.oreilly.com/library/view/-/9780136083238/?ar

File: manual/guides/pull_requests.md [hunk 13]
```diff
@@ -0,0 +1,96 @@
+---
+layout: manual
+title: Pull Requests
+subtitle: "The #1 Collaboration Tool"
+permalink: /manual/guides/pull_requests
+---
+
+# Making a Pull Request
+
+A Pull Requests (PR) merges the changes in a branch of a repository
```
Comment (katyhuff) [inline manual/guides/pull_requests.md:10]:
```suggestion
A Pull Request (PR) merges the changes in a branch of a repository
```

File: manual/guides/pull_requests.md [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+to the main branch. This means the main branch has all of the updates
+that need to get approved before they're implemented.
+For the basics on how to create a PR, GitHub has a guide available
+[here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request).
+
+Most of the pull requests done in the ARFC group are from a working branch
```
Comment (katyhuff) [inline manual/guides/pull_requests.md:16]:
```suggestion
Most of the pull requests done in the ARFC group are from a feature branch 
(not the main branch)
```

File: manual/guides/pull_requests.md [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+in a personal fork to the main ARFC repository. PRs made by
```
Comment (katyhuff) [inline manual/guides/pull_requests.md:17]:
```suggestion
in a personal fork to the main branch in the ARFC fork of a repository. 
PRs made in the ARFC organization should:
```

File: manual/guides/pull_requests.md [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+ARFC members should:
```
Comment (katyhuff) [inline manual/guides/pull_requests.md:18]:
```suggestion
```

File: manual/guides/pull_requests.md [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+      close this issue when the PR is merged
+- [ ] Thank the author for their contribution.
+```
+
+# Responding to Pull Requests Reviews
+
+Reviews on pull requests includes comments, approvals, and
+suggestions for changes. Comments can include general comments on the
+material or the formatting of the submission. Changes can be
+requested for a specific line or lines, where the reviewer will write
```
Comment (katyhuff) [inline manual/guides/pull_requests.md:90]:
```suggestion
requested for a specific line or lines, where the reviewer can even write
```

File: manual/guides/pull_requests.md [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+their own code to improve the committed files. These changes can
+either be ignored or committed directly from the review. The
```
Comment (katyhuff) [inline manual/guides/pull_requests.md:92]:
```suggestion
either be discussed in the comment box or committed directly from the
review with the `commit suggestion` button. With this button, the
```

File: manual/guides/pull_requests.md [hunk 19]
```diff
@@ -0,0 +1,96 @@
...
+committed changes are implemented immediately in GitHub, which makes
```
Comment (katyhuff) [inline manual/guides/pull_requests.md:93]:
```suggestion
suggested changes are committed and pushed immediately
to the PR branch, which makes
```

File: manual/guides/pull_requests.md [hunk 20]
```diff
@@ -0,0 +1,96 @@
...
+incorporating suggested changes faster. Once the PR has been approved
+by the reviewers, one reviewer will merge the changes and the PR will
```
Comment (katyhuff) [inline manual/guides/pull_requests.md:95]:
```suggestion
by at least one reviewer, a reviewer will merge the changes and the PR will
```

File: manual/guides/pull_requests.md [hunk 21]
```diff
@@ -0,0 +1,96 @@
...
+        in your review?
+- [ ] When you approve of the PR, merge and close it.
+- [ ] Does this PR close an issue? If so, be sure to descriptively
+      close this issue when the PR is merged
+- [ ] Thank the author for their contribution.
+```
+
+# Responding to Pull Requests Reviews
+
+Reviews on pull requests includes comments, approvals, and
```
Comment (katyhuff) [inline manual/guides/pull_requests.md:87]:
```suggestion
Reviews on pull requests include comments, approvals, and
```

File: manual/guides/pull_requests.md [hunk 22]
```diff
@@ -0,0 +1,96 @@
...
+suggestions for changes. Comments can include general comments on the
+material or the formatting of the submission. Changes can be
```
Comment (katyhuff) [inline manual/guides/pull_requests.md:89]:
```suggestion
technical substance, documentation, performance, clarity, reproducibility, 
coding style, and even the formatting of the submission. Changes can be
```

File: manual/guides/pull_requests.md [hunk 23]
```diff
@@ -0,0 +1,96 @@
...
+requested for a specific line or lines, where the reviewer will write
+their own code to improve the committed files. These changes can
+either be ignored or committed directly from the review. The
+committed changes are implemented immediately in GitHub, which makes
+incorporating suggested changes faster. Once the PR has been approved
```
Comment (katyhuff) [inline manual/guides/pull_requests.md:94]:
```suggestion
incorporating suggested changes faster. Do not leave comments un-handled.
All reviewer comments should be incorporated, responded to, handled, or 
discussed further. Once the PR has been approved
```
